### PR TITLE
Introducting LLM debug mode

### PR DESCRIPTION
**Pull request by rakeshv247**
## Summary

The pull request adds changes to introduce the debug mode to the `StartConversation` message. Currently, the debug mode only supports llm interactions. If set, the client should receive all the llm interaction between flow-service and LLM backends.

This feature mainly monitors and troubleshoots LLM interactions as and when needed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [0.0.10] - 2025-02-13
```
Comment (clairesc) [inline CHANGELOG.md:7]: I know we're not doing a great job at following semantic versioning rules, but this time I would bump to 0.1.0 as it's introducing support for a new feature, not just bugfixing existing code.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+
+- Added Debug mode
+  - Introduced a new debug field in `StartConversation`.
+  - When debug mode is enabled, the client can receive LLM interaction messages exchanged between the flow service and LLM backends.
```
Comment (clairesc) [inline CHANGELOG.md:13]:
```suggestion
  - When LLM debug mode is enabled, the client will receive debug messages with LLM request and response content
```
I would rephrase with something more like that

File: speechmatics_flow/client.py [hunk 3]
```diff
@@ -586,6 +593,9 @@ async def run(
         :param playback_settings: Configuration for the playback stream.
         :type playback_settings: models.PlaybackSettings
 
+        :param debug_mode: Configuration to receive debug messages from flow-service
```
Comment (clairesc) [inline speechmatics_flow/client.py:596]:
```suggestion
        :param debug_mode: Configuration to receive debug messages from Flow

File: speechmatics_flow/models.py [hunk 4]
```diff
@@ -91,6 +91,18 @@ def asdict(self):
         )
 
 
+@dataclass
+class DebugMode:
+    """Defines debug flags for monitoring and troubleshooting flow internals"""
```
Comment (clairesc) [inline speechmatics_flow/models.py:96]:
```suggestion
    """Defines debug flags for monitoring and troubleshooting Flow"""
```

File: speechmatics_flow/models.py [hunk 5]
```diff
@@ -91,6 +91,18 @@ def asdict(self):
...
+
+    llm: bool = False
+    """Optional Flag indicating whether to capture conversations between the LLM and flow-service in
```
Comment (clairesc) [inline speechmatics_flow/models.py:99]:
```suggestion
    """Optional Flag indicating whether to receive conversations between the LLM and the Flow backend as
```

File: speechmatics_flow/models.py [hunk 6]
```diff
@@ -91,6 +91,18 @@ def asdict(self):
...
+@dataclass
+class DebugMode:
+    """Defines debug flags for monitoring and troubleshooting flow internals"""
+
+    llm: bool = False
+    """Optional Flag indicating whether to capture conversations between the LLM and flow-service in 
+    debug messages."""
+
+    def asdict(self):
+        return {"llm": self.llm}
```
Comment (clairesc) [inline speechmatics_flow/models.py:103]: This could be more generic, the same as `ConversationConfig` `asdict`, or simpler for now as we just have a bool with default, so
```
def asdict(self):
    return asdict(self)
```

File: speechmatics_flow/models.py [hunk 7]
```diff
@@ -179,7 +191,10 @@ class ServerMessageType(str, Enum):
     """Indicates a generic warning message."""
 
     Error = "Error"
-    """Indicates n generic error message."""
+    """Indicates a generic error message."""
+
+    Debug = "Debug"
+    """Indicates a debug message based on the flag set in the StartConversation message"""
```
Comment (clairesc) [inline speechmatics_flow/models.py:197]:
```suggestion
    """Indicates a debug message"""
```
I know it doesn't sound very descriptive, but I would l just say this. We have undocumented `Debug` messages already going over the websocket that cannot be configured from `StartConversation`

File: tests/test_client.py [hunk 8]
```diff
@@ -58,13 +62,27 @@ def ws_client():
...
+            id="with default values and llm debug mode enabled",
+        ),
     ],
 )
 def test_start_conversation(
     ws_client: WebsocketClient,
     audio_format: AudioSettings,
     conversation_config: ConversationConfig,
     tools: Optional[List[ToolFunctionParam]],
+    debug_mode,
```
Comment (clairesc) [inline tests/test_client.py:85]:
```suggestion
    debug_mode: Optional[DebugMode],
```

File: tests/test_models.py [hunk 9]
```diff
@@ -57,3 +57,16 @@ def test_playback_settings(config, want):
     audio_settings = models.PlaybackSettings(**config)
     got = asdict(audio_settings)
     assert got == want
+
+
+@pytest.mark.parametrize(
+    "config, want",
+    [
+        ({"llm": False}, {"llm": False}),
+        ({"llm": True}, {"llm": True}),
```
Comment (clairesc) [inline tests/test_models.py:66]: Add a test with empty input that expect `{"llm": False}`?

File: CHANGELOG.md (hunk 1 above)
Comment (rakeshv247) [reply CHANGELOG.md]: done in https://github.com/speechmatics/speechmatics-flow/pull/12/commits/d22553cbbc50f64b10bf2780a28e6620e706c974

File: speechmatics_flow/models.py (hunk 6 above)
Comment (rakeshv247) [reply speechmatics_flow/models.py]: done in https://github.com/speechmatics/speechmatics-flow/pull/12/commits/ed5088c7c7f782aa3437308ded1a2e4ee45efa1f

File: tests/test_models.py (hunk 9 above)
Comment (rakeshv247) [reply tests/test_models.py:67]: done in https://github.com/speechmatics/speechmatics-flow/pull/12/commits/0676471102bfaa786ec8a81ee381f0b297fb47e5

File: speechmatics_flow/cli_parser.py [hunk 10]
```diff
@@ -153,6 +153,12 @@ def get_arg_parser():
         choices=[k for k in TemplateOptions.keys()],
         help="Choose your assistant.",
     )
+    parser.add_argument(
+        "--llm-debug-enabled",
+        default=False,
+        action="store_true",
+        help="Flag indicating whether to capture conversations between the LLM and flow-service in debug messages.",
```
Comment (clairesc) [inline speechmatics_flow/cli_parser.py:160]:
```suggestion
        help="Flag indicating whether to receive conversations between the LLM and the Flow backend in debug messages.",
```
Missed this one earlier.

**Review by clairesc — APPROVED** [review]
